adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureRDDFunctions.scala

```diff
@@ -133,11 +132,167 @@ class FeatureRDDFunctions(featureRDD: RDD[Feature]) extends Serializable with Lo
     genes
   }
 
+  def reassignParentIds(): RDD[Feature] = {
+    def reassignParentId(f: Feature): Feature = {
+      val fb = Feature.newBuilder(f)
+
+      val featureId: Option[String] = Option(f.getFeatureId)
+      val geneId: Option[String] = Option(f.getGeneId)
+      val transcriptId: Option[String] = Option(f.getTranscriptId)
+      val exonNumber: Option[String] = Option(f.getAttributes.get("exon_number"))
+      val exonId: Option[String] = Option(f.getExonId) match {
+        case Some(e) => Option(e)
+        case None    => transcriptId.flatMap(t => exonNumber.map(e => t + "_" + e))
+      }
+
+      val (_featureId, _parentId) =
```

Comment:
We don't use `_` prefixes elsewhere in this codebase, so I'd prefer not to introduce them here. I would suffix `featureId: Option[String]` to `featureIdOpt`, which is fairly common across Scala projects.